app/src/main/java/de/danoeh/antennapod/activity/SelectSubscriptionActivity.java

```diff
@@ -0,0 +1,162 @@
+        setTheme(UserPreferences.getTranslucentTheme());
+        super.onCreate(savedInstanceState);
+
+        viewBinding = SubscriptionSelectionActivityBinding.inflate(getLayoutInflater());
+        setContentView(viewBinding.getRoot());
+        setSupportActionBar(viewBinding.toolbar);
+        setTitle(R.string.shortcut_select_subscription);
+
+        viewBinding.transparentBackground.setOnClickListener(v -> finish());
+        viewBinding.card.setOnClickListener(null);
+
+        loadSubscriptions();
+
+        final Integer[] checkedPosition = new Integer[1];
+        viewBinding.list.setChoiceMode(ListView.CHOICE_MODE_SINGLE);
+        viewBinding.list.setOnItemClickListener((listView, view1, position, rowId) ->
+                checkedPosition[0] = position
+        );
+        viewBinding.shortcutBtn.setOnClickListener(view -> {
+            if (checkedPosition[0] != null && Intent.ACTION_CREATE_SHORTCUT.equals(
+                    getIntent().getAction())) {
+                getBitmapFromUrl(listItems.get(checkedPosition[0]));
+            }
+        });
+
+    }
+
+    public List<Feed> getFeedItems(List<NavDrawerData.DrawerItem> items, List<Feed> result) {
+        for (NavDrawerData.DrawerItem item : items) {
+            if (item.type == NavDrawerData.DrawerItem.Type.TAG) {
+                getFeedItems(((NavDrawerData.TagDrawerItem) item).children, result);
+            } else {
+                Feed feed = ((NavDrawerData.FeedDrawerItem) item).feed;
+                if (!result.contains(feed)) {
+                    result.add(feed);
+                }
+            }
+        }
+        return result;
+    }
+
+    private void addShortcut(Feed feed, Bitmap bitmap) {
+        Intent intent = new Intent(this, MainActivity.class);
+        intent.setAction(Intent.ACTION_MAIN);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        intent.putExtra(EXTRA_FEED_ID, feed.getId());
+        String id = "subscription-" + feed.getId();
+        IconCompat icon;
+
+        if (bitmap != null) {
+            icon = IconCompat.createWithAdaptiveBitmap(bitmap);
+        } else {
+            icon = IconCompat.createWithResource(this, R.drawable.ic_folder_shortcut);
+        }
+
+        ShortcutInfoCompat shortcut = new ShortcutInfoCompat.Builder(this, id)
+                .setShortLabel(feed.getTitle())
+                .setLongLabel(feed.getFeedTitle())
+                .setIntent(intent)
+                .setIcon(icon)
+                .build();
+
+        setResult(RESULT_OK, ShortcutManagerCompat.createShortcutResultIntent(this, shortcut));
+        finish();
+    }
+
+    private void getBitmapFromUrl(Feed feed) {
+        int iconSize = (int) (128 * getResources().getDisplayMetrics().density);
+        Glide.with(this)
+                .asBitmap()
+                .load(feed.getImageUrl())
+                .apply(new RequestOptions().override(iconSize, iconSize))
```

Comment:
When using the full size image bitmap I was running into ` FAILED BINDER TRANSACTION` error, so resized the image not sure of the icon size so used the notification icon size.

Reply:
Should be fine :) Just tried it again. The code looks good¹ and works fine. Will merge when the CI check is green.

¹ The `Integer[]` looks a bit weird but I guess we can keep it